### PR TITLE
fix run_squad.py during fine-tuning xlnet on squad2.0

### PR DESCRIPTION
**Pull request by importpandas**
The following is a piece of code in forward function of xlnet model, which obviously is the key point of training the model on unanswerable questions using cls token representations.  But the default value of tensor `is_impossible`(using to indicate whether this example is answerable) is none, and we also hadn't passed this tensor into forward function. That's the problem. 

```        
 if cls_index is not None and is_impossible is not None:
                # Predict answerability from the representation of CLS and START
                cls_logits = self.answer_class(hidden_states, start_positions=start_positions, cls_index=cls_index)
                loss_fct_cls = nn.BCEWithLogitsLoss()
                cls_loss = loss_fct_cls(cls_logits, is_impossible)
                total_loss += cls_loss * 0.5
```

I added the `is_impossible` tensor to TensorDataset and model inputs, and got a reasonable result, 

{
"exact": 80.4177545691906,
"f1": 84.07154997729623,
"total": 11873,
"HasAns_exact": 77.59784075573549,
"HasAns_f1": 84.83993323200234,
"HasAns_total": 5928,
"NoAns_exact": 84.0874684608915,
"NoAns_f1": 84.0874684608915,
"NoAns_total": 5945
}

My running command:
```
python run_squad.py \
    --model_type xlnet \
    --model_name_or_path xlnet-large-cased \
    --do_train \
    --do_eval \
    --version_2_with_negative \
    --train_file data/train-v2.0.json \
    --predict_file data/dev-v2.0.json \
    --learning_rate 3e-5 \
    --num_train_epochs 4 \
    --max_seq_length 384 \
    --doc_stride 128 \
    --output_dir ./wwm_cased_finetuned_squad/ \
    --per_gpu_eval_batch_size=2  \
    --per_gpu_train_batch_size=2   \
    --save_steps 5000
```
I run my code with 4 nvidia GTX 1080ti gpus and pytorch==1.2.0.

### CODE REVIEW

**Comment by thomwolf** [comment]
This looks good, do you want to add your command and the results you mention in the README of the examples in `examples/README.md`?

**Comment by ShengeBelmendo** [comment]
ok, I have updated the readme.

**Comment by ahotrod** [comment]
@ShengeBelmendo thanks for contributing this fix.  My results running this mod:
```
{
  "exact": 40.93320980375642,
  "f1": 45.34019049783724,
  "total": 11873,
  "HasAns_exact": 81.81511470985156,
  "HasAns_f1": 90.64171420054345,
  "HasAns_total": 5928,
  "NoAns_exact": 0.16820857863751051,
  "NoAns_f1": 0.16820857863751051,
  "NoAns_total": 5945
}
```
which is nearly identical to results from `run_squad.py` without `is_impossible`.  My run command for bs=48 using gradient accumulation consumes 10975 MiB on each of 2x NVIDIA 1080Ti:
```
#!/bin/bash
export OMP_NUM_THREADS=6
RUN_SQUAD_DIR=/media/dn/dssd/nlp/transformers/examples
SQUAD_DIR=${RUN_SQUAD_DIR}/scripts/squad2.0
MODEL_PATH=${RUN_SQUAD_DIR}/runs/xlnet_large_squad2_512_bs48_II

python -m torch.distributed.launch --nproc_per_node=2 ${RUN_SQUAD_DIR}/run_squad_II.py \
  --model_type xlnet \
  --model_name_or_path xlnet-large-cased \
  --do_train \
  --train_file ${SQUAD_DIR}/train-v2.0.json \
  --predict_file ${SQUAD_DIR}/dev-v2.0.json \
  --version_2_with_negative \
  --num_train_epochs 3 \
  --learning_rate 3e-5 \
  --adam_epsilon 1e-6 \
  --max_seq_length 512 \
  --doc_stride 128 \
  --save_steps 2000 \
  --per_gpu_train_batch_size 1 \
  --gradient_accumulation_steps 24 \
  --output_dir ${MODEL_PATH}

CUDA_VISIBLE_DEVICES=0 python ${RUN_SQUAD_DIR}/run_squad_II.py \
  --model_type xlnet \
  --model_name_or_path ${MODEL_PATH} \
  --do_eval \
  --train_file ${SQUAD_DIR}/train-v2.0.json \
  --predict_file ${SQUAD_DIR}/dev-v2.0.json \
  --max_seq_length 512 \
  --per_gpu_eval_batch_size 48 \
  --output_dir ${MODEL_PATH}
$@
```
```
Saving features into cached file /media/dn/dssd/nlp/transformers/examples/scripts/squad2.0/cached_train_xlnet-large-cased_512
***** Running training *****
Num examples = 130738
Num Epochs = 3
Instantaneous batch size per GPU = 1
Total train batch size (w. parallel, distributed & accumulation) = 48
Gradient Accumulation steps = 24
Total optimization steps = 8169
```
Could anything related to distributed processing be responsible for this result?  Batch size 48 per the original paper too aggressive?  Not running fp16, so I believe there are no issues with gradients out of range, and I get no warnings.  I have time to try other configurations, what would you suggest?  My run took 34+ hours so a targeted approach for narrowing this down is warranted, duh ...

**Comment by ShengeBelmendo** [comment]
@ahotrod As for your first results, I will suggest you watch my changes on file `run_squad.py` in this pr. It just added a another input to the model and only a few lines. Without this changing, I got the same results with you, which is nearly to 0 on unanswerable questions. I think Distributed processing isn't responsible for that. When it comes to the running script, it doesn't matter since I had hardly changed it.

**Comment by ahotrod** [comment]
@importpandas re-run just completed (36 hours) and results are posted above.
Works great, thanks again!

**Comment by importpandas** [comment]
> @importpandas re-run just completed (36 hours) and results are posted above.
> Works great, thanks again!

okay, pleasure

**Comment by thomwolf** [comment]
LGTM, merging cc @LysandreJik (the `run_squad` master)
